src/Xamarin.Android.Build.Utilities/Sdks/AndroidSdkBase.cs

```diff
@@ -81,6 +81,32 @@ public virtual void Initialize (string androidSdkPath = null, string androidNdkP
 				IsNdk64Bit = Directory.EnumerateDirectories (toolchainsDir, "arm-linux-androideabi-*")
 					.Any (dir => Directory.Exists (Path.Combine (dir, "prebuilt", NdkHostPlatform64Bit)));
 			}
+			// we need to look for extensions other than the default .exe|.bat
+			// google have a habbit of changing them.
+			foreach (var exe in Executables (Adb)) {
+				if (File.Exists (Path.Combine (AndroidPlatformToolsPath, exe)))
+					Adb = exe;
```

Comment:
Aside: do we remember why `Adb` isn't a full path, but instead just a filename? We have so many different directories running around...

Making such a change would likely break too much existing code, so this might be a meaningless question, but really, what were we thinking?!